stock_quant_cost_segmentation/model/stock.py

```diff
@@ -109,12 +109,28 @@ def product_segmentation_update_after_done(
 class StockQuant(models.Model):
     _inherit = "stock.quant"
 
-    @api.depends('material_cost', 'production_cost', 'landed_cost',
-                 'subcontracting_cost')
-    def _compute_segmentation(self):
-        for record in self:
-            record.segmentation_cost = sum([
-                getattr(record, fn) for fn in SEGMENTATION_COST])
+    @api.model
+    def create(self, vals):
+        segmentation_cost = sum([
+            getattr(self, field_name) for field_name in SEGMENTATION_COST])
+        vals.update({'segmentation_cost': segmentation_cost})
+        return super(StockQuant, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        res = super(StockQuant, self).write(vals)
+        if not set(SEGMENTATION_COST).issubset(set(vals)) or \
+           not self.env.context.get('force_segmentation_cost'):
+            return res
+        # TODO: Validate sql injection from SEGMENTATION_COST variable
+        # Because other module could add a monkey patch with sql injection
+        sum_query = ' + '.join(["COALESCE(%s, 0)" % field_name
+                                for field_name in SEGMENTATION_COST])
+        # NOTE: Cache is too slow to manage many records from computed field
+        query = "UPDATE stock_quant SET segmentation_cost = (" + sum_query + \
+            ") WHERE id IN %s"
```

Comment:
I think in order to avoid a write we should have a little IF:

IF computation is equal to what the field already has then do not write.